src/hotspot/share/opto/c2_globals.hpp

```diff
@@ -660,6 +660,16 @@
   develop(bool, VerifyAliases, false,                                       \
           "perform extra checks on the results of alias analysis")          \
                                                                             \
+  develop(uint, VerifyConstraintCasts, 0,                                   \
+          "Perform runtime checks to verify the value of a "                \
+          "ConstraintCast lies inside its type"                             \
+          "0=does not perform any verification, "                           \
+          "1=perform verification on ConstraintCastNodes that are present " \
+          "during code emission, "                                          \
+          "2=Do not do widening of ConstraintCastNodes so that we can "     \
+          "have more verification coverage")                                \
+          range(0, 2)                                                       \
+                                                                            \
```

Comment:
Nit: I think the alignment could be a little improved, maybe also the comments.
Here a quick draft:

```suggestion
  develop(uint, VerifyConstraintCasts, 0,                                   \
          "Perform runtime checks to verify the value of a ConstraintCast " \
          "lies inside its type"                                            \
          "0: No verification."                                             \
          "1: Verify (types are widened, same as without verification)."  \
          "2: Verify, where types are not widened, for better  " \
          "   verification coverage"). \
          range(0, 2)                                                       \
                                                                            \
```
I'm too lazy to fix the alignment of the `\`. This here is just a suggestion, I leave it up to you in the end ;)

Reply:
Done, I believe the common theme in this file is to use `=` to signify the meanings of different values of the same variable while `:` is to signify the meanings of different variables.

E.g:

      develop(uint, VerifyIterativeGVN, 0,                                      \
              "Verify Iterative Global Value Numbering"                         \
              "=XY, with Y: verify Def-Use modifications during IGVN"           \
              "          X: verify that type(n) == n->Value() after IGVN"       \
              "X and Y in 0=off; 1=on")                                         \
              constraint(VerifyIterativeGVNConstraintFunc, AtParse)             \